### PR TITLE
Adds code to remove extra directories when uploading archives.

### PR DESCRIPTION
**Pull request by james-harder**
## Checklist

🚨 Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository. 🚨

- [x] Make sure you are making a pull request against our **main branch** (left side)
- [x] Check that that your branch is up to date with our main.
- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request *your* main!
- [x] Check that the tests and code linter both pass. See [1]
- [x] If you're a new contributor, please sign our [contributor license agreement](https://cla-assistant.io/manyfold3d/manyfold).

## Warnings

- [ ] This PR will change existing database contents.
- [ ] This PR introduces a breaking change to existing installations.

## Summary

This PR is meant to resolve the issue where archives containing directories are creating extraneous directories in the Library once uploaded.

## Linked issues

resolves #{1705}

## Description of changes

This change works by using the `entry.pathname=` method from the ffi-libarchive gem to strip the path from the archived file name. The result is that all files in the archive will be extracted to a directory with the same name as the archive.

[1]
I'm seeing the following error from `bundle exec rubocop`. This is a method being called, not a variable being set, so I'm not sure the best way to deal with the rubocop error.

```bash
app/controllers/uploads_controller.rb:43:25: C: [Correctable] Layout/SpaceAroundOperators: Surrounding space missing for operator =.
          entry.pathname=(file_name.to_s)
```

### CODE REVIEW

File: app/controllers/uploads_controller.rb [hunk 1]
```diff
@@ -36,7 +36,13 @@ def unzip(dest_folder_name, datafile)
     Dir.mkdir(dest_folder_name)
     Dir.chdir(dest_folder_name) do
       reader.each_entry do |entry|
-        reader.extract(entry, flags.to_i)
+        # For non-directory files, strips the path and extracts to current directory
```
Comment (Floppy) [inline app/controllers/uploads_controller.rb:39]: Will this flatten *all* directory structure inside the zipfile, or just the top level?

Comment (james-harder) [reply app/controllers/uploads_controller.rb:39]: This flattens all directory structure.

Comment (Floppy) [reply app/controllers/uploads_controller.rb:39]: Hm - lots of models have subdirectories that people want to keep. We should just drop the top-level directory if there's just a single one there.

Comment (james-harder) [reply app/controllers/uploads_controller.rb]: Updated code to only change the structure if there is a single directory at the root of the archive. Added four archives to use for testing.

**Comment by Floppy** [comment]
This looks good. Are those zipfiles used in some tests that you've not added to the PR? If they're just for interactive testing, probably best to leave them out. I've got some ideas how we could make some dynamically for testing anyway (I'm paranoid about binary blobs in git after the xz thing the other week).

**Comment by james-harder** [comment]
I don't have a lot of experience writing tests, so I was just using them to test interactively. I went ahead and took them out, though :)

**Comment by Floppy** [comment]
Thinking about it, this isn't currently tested anyway, and I'm hopefully embarking on a bigger rewrite of the uploader generally soon, so I'll merge as-is and we'll get tests on it later.

**Pull request merged**